porcelain/storagedeals.go

```diff
@@ -140,3 +142,84 @@ func filterDealChannel(dealCh <-chan *StorageDealLsResult, filterFunc func(*stor
 
 	return outCh
 }
+
+type dealRedeemPlumbing interface {
+	ChainBlockHeight() (*types.BlockHeight, error)
+	DealGet(context.Context, cid.Cid) (*storagedeal.Deal, error)
+	MessagePreview(context.Context, address.Address, address.Address, string, ...interface{}) (types.GasUnits, error)
+	MessageSendWithDefaultAddress(context.Context, address.Address, address.Address, *types.AttoFIL, types.AttoFIL, types.GasUnits, string, ...interface{}) (cid.Cid, error)
+}
+
+// DealRedeem redeems a voucher for the deal with the given cid and returns
+// either the cid of the created redeem message or an error
+func DealRedeem(ctx context.Context, plumbing dealRedeemPlumbing, fromAddr address.Address, dealCid cid.Cid, gasPrice types.AttoFIL, gasLimit types.GasUnits) (cid.Cid, error) {
```

Comment:
This plumbing function needs a test to confirm it correctly chooses the best voucher from a deal and sends a properly formatted message.